packages/consul/src/Consul.js

```diff
@@ -169,13 +171,17 @@ export class Consul {
 
     const list = await retry(async () => this.consulInstance.agent.service.list(), retryOptions);
 
-    const isRegistered = Object.entries(list).some(([id, { Service }]) => id === data.id && Service === data.name);
+    const isAlreadyRegistered = Object.entries(list).some(([id, { Service }]) => id === data.id && Service === data.name);
 
-    if (!isRegistered) {
+    if (!isAlreadyRegistered) {
       await retry(async () => this.consulInstance.agent.service.register(data), retryOptions);
 
       this.registerParams.id = data.id;
     }
+
+    if (options.onAfterRegister) {
+      options.onAfterRegister(isAlreadyRegistered);
+    }
```

Comment:
```suggestion
      options?.onAfterRegister?.(isAlreadyRegistered);
```

Reply:
Not sure that we already have it in the JS, like in TS, but I will check it. Thanks for the idea 🙏

Reply:
Oh, yes, we do have it 👍